api/client.go

```diff
@@ -287,3 +292,19 @@ func (c *Client) Heartbeat(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (c *Client) CreateLayerRequest(ctx context.Context, digest string, r io.Reader) (string, error) {
+	var response CreateLayerResponse
+	if err := c.do(ctx, http.MethodGet, fmt.Sprintf("/api/layers/%s", digest), nil, &response); err != nil {
```

Comment:
This can't be a HEAD request since it needs to GET the path. Not only does this need to check if the blob exists, it needs to also return the server-side file path